analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.Arithmetic.cs

```diff
@@ -154,4 +168,45 @@ private static NumberConstraint CalculateAnd(NumberConstraint left, NumberConstr
             return BigInteger.Max(left.Min.Value, right.Min.Value);
         }
     }
+
+    private static NumberConstraint CalculateOr(NumberConstraint left, NumberConstraint right) =>
+        NumberConstraint.From(CalculateOrMin(left, right), CalculateOrMax(left, right));
+
+    private static BigInteger? CalculateOrMin(NumberConstraint left, NumberConstraint right)
+    {
+        // BitOr can only turn 0s into 1s, not the other way around => If both operands have the same sign, the result cannot be smaller than the bigger of the two.
+        if ((left.IsNegative && right.IsNegative) || (left.IsPositive && right.IsPositive))
+        {
+            return BiggestMinimum(left, right);
+        }
+        // -1 == 0b11111111 => a | -1 == -1 =>
+        // If one operand can be -1 and one is positive, the result will be negative, but not smaller than the negative operand.
+        else if (left.IsNegative)
```

Comment:
As discussed, and for the record: yes, the result won't be smaller than the minimum of that range.
However, we don't just settle with that result in general:
- we have already dealt with the cases `(L negative, R negative)` and `(L positive, R positive)` in the 1st branch of the if;
- so in this branch of the `if` we have `(L negative, R positive)` or `(L negative, R mixed)`;
- it turns out that, the min for both cases is the min of L (check the new version for the proof);
- in the following branch of the `if` we have `(L positive, R negative)` or `(L mixed, R negative)`;
- applying the same logic on R, instead of L, we conclude that the min for both cases is the min of R;
- in the last branch of the `if` we have one negative and the other mixed, or both mixed;
- in such scenarios, we resort to the property "result won't be smaller than the minimum of that range".